File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
import datetime
from dateutil.relativedelta import relativedelta

def solution(today, terms, privacies):
    answer = []
    format = '%Y-%m-%d'
    dict = {}
    today = today.replace('.','-')
    dt_datetime = datetime.datetime.strptime(today,format)
    
    for i in terms:
        a,b = i.split()
        dict[a] = int(b)
    
    for i in range(len(privacies)):
        a , b= privacies[i].split()
        # b = int(b)
        
        p_day = a.replace('.','-')
        p = datetime.datetime.strptime(p_day,format)
        # pday = p+datetime.timedelta(days=dict[b] * 28)
        plus = p + relativedelta(months=dict[b])
        print(plus)
        if plus <= dt_datetime:
            answer.append(i+1)
        
    
    
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
    var answer = [];
    
//     A => 6달
//     B => 12달
//     C => 3달
    let [t_year, t_month, t_day] = today.split('.').map(Number);
    let now =( t_year * 12 * 28 ) + (28 * t_month) + t_day;
    
    let obj = {}
    
    for(let t of terms){
        const [type,m] = t.split(' ');
        obj[type] = parseInt(m) * 28;
    }
    console.log(obj)
    
    for(let  i=0; i < privacies.length; i ++){
        let [date, type] = privacies[i].split(' ');
        let [year,month,day] = date.split('.').map(Number);
        const priv_day = (year * 12 * 28) + (month * 28) + day + obj[type];
        if (priv_day <= now){
            answer.push(i+1)
        }
    }
    return answer;
}